File: 4. greedy/greedy.py
############################################################################
# greedy.py
#
# Minor Programming
# Programming II
# Problem Set 6: Sentimental
# Najib el Moussaoui
#
# Returns change owed in amount of coins.
###########################################################################

import cs50

# prompt user for amount owed
while True:
    dollars = cs50.get_float("How much change is owed (in dollars)? ")
    if dollars > 0:
        break

# convert dollars to cents and initialize counters
cents = dollars * 100
amount = cents
count = 0

# iterate over amount, substract and adjust counters
while True:
    if amount >= 25:
        amount -= 25
        count += 1
    if amount >= 10 and amount < 25:
        amount -= 10
        count += 1
    if amount >= 5 and amount < 10:
        amount -= 5
        count += 1
    if amount >= 1 and amount < 5:
        amount -= 1
        count += 1
    if amount == 0:
        break

# print amount in coins
print(f"Amount in coins: {count}")

File: 4. greedy/greedy.c
/****************************************************************************
 * greedy.c
 *
 * Minor Programming
 * Najib el Moussaoui
 *
 * Convert amount of change owed into coins.
 ***************************************************************************/
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    float dollars;

    do
    {
        // prompt user for change owed in dollars
        printf("How much change is owed? ");
        dollars = get_float();
    }
    while (dollars < 0);

    // convert dollars to cents and round off
    float cents = round(dollars * 100);

    int amount = cents;

    // counter to keep track of amount of coints
    int count = 0;

    // substract quarters from amount, increment coin counter
    if (amount >= 25)
    {
        do
        {
            amount -= 25;
            count++;
        }
        while (amount >= 25);
    }
    // substract dimes from amount, increment coin counter
    if (amount >= 10)
    {
        do
        {
            amount -= 10;
            count++;
        }
        while (amount >= 10);
    }
    // substract nickels from amount, increment coin counter
    if (amount >= 5)
    {
        do
        {
            amount -= 5;
            count++;
        }
        while (amount >= 5);
    }
    // substract pennies from amount, increment coin counter
    if (amount >= 1)
    {
        do
        {
            amount -= 1;
            count++;
        }
        while (amount >= 1);
    }
    // print amount of coins to the console
    printf("Amount of coins: %i\n", count);
}